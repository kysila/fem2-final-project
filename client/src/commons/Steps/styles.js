import { makeStyles } from '@material-ui/core';

export const useItemStyles = makeStyles({
  item: {
    color: (props) => (props.isActive ? '#444444' : '#AAAAAA !important'),
  },
});

export const useStyles = makeStyles(() => ({
  container: {},
  separator: {
    color: '#AAAAAA',
    fontSize: '9px',
    margin: '0 9px',
  },
}));
