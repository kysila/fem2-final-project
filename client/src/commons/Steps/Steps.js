import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Box, Typography } from '@material-ui/core';

import ArrowForwardIosIcon from '@material-ui/icons/ArrowForwardIos';
import MuseoSans from '../../fonts/MuseoSans-500.woff';

import { useStyles, useItemStyles } from './styles';

const museo = {
  fontFamily: 'Museo Sans 500',
  fontStyle: 'normal',
  fontSize: '13px',
  src: `
    local('Museo Sans 500'),
    url(${MuseoSans}) format('woff')
  `,
};

export function Separator(props) {
  return (
    <Box component="span"><ArrowForwardIosIcon className={props.classes.separator} /></Box>
  );
}

export function StepsItem(props) {
  const classes = useItemStyles(props);

  return (
    <Typography
      style={museo}
      component="span"
      className={classes.item}
    >
      {props.text}
    </Typography>
  );
}

export function Steps(props) {
  const classes = useStyles(props);

  return (
    <Box className={classNames(props.className, classes.container)}>
      {props.items.reduce(
        (result, item) => result.concat([
          <StepsItem key={item.key} {...item} />,
          <Separator key={item.key.concat('s')} classes={classes} />,
        ]), [],
      ).slice(0, -1)}
    </Box>
  );
}
