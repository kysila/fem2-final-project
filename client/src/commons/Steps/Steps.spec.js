/* eslint-disable */
import React from 'react';
import {shallow} from 'enzyme';
import {Steps} from './Steps';

const variants = [
  [
    {
      key: 'id-0',
      isActive: true,
      text: 'Cart',
    },
    {
      key: 'id-1',
      isActive: false,
      text: 'Information',
      src: '/checkout/info',
    },
    {
      key: 'id-2',
      isActive: false,
      text: 'Shipping',
      src: null,
    },
    {
      key: 'id-3',
      isActive: false,
      text: 'Payment',
      src: null,
    },
  ],
  [
    {
      key: 'id-0',
      isActive: true,
      text: 'Cart',
    },
    {
      key: 'id-1',
      isActive: true,
      text: 'Information',
      src: '/checkout/info',
    },
    {
      key: 'id-2',
      isActive: false,
      text: 'Shipping',
      src: null,
    },
    {
      key: 'id-3',
      isActive: false,
      text: 'Payment',
      src: null,
    },
  ],
  [
    {
      key: 'id-0',
      isActive: true,
      text: 'Cart',
    },
    {
      key: 'id-1',
      isActive: true,
      text: 'Information',
      src: '/checkout/info',
    },
    {
      key: 'id-2',
      isActive: true,
      text: 'Shipping',
      src: '/checkout/shipping',
    },
    {
      key: 'id-3',
      isActive: false,
      text: 'Payment',
      src: null,
    },
  ],
  [
    {
      key: 'id-0',
      isActive: true,
      text: 'Cart',
    },
    {
      key: 'id-1',
      isActive: true,
      text: 'Information',
      src: '/checkout/info',
    },
    {
      key: 'id-2',
      isActive: true,
      text: 'Shipping',
      src: '/checkout/shipping',
    },
    {
      key: 'id-3',
      isActive: true,
      text: 'Payment',
      src: '/checkout/payment',
    },
  ],
];

describe('Steps component', () => {
  variants.forEach((items) => {
    it('Steps render', () => {
      const rendered = shallow(
        <Steps items={items} />,
      );

      expect(rendered).toMatchSnapshot();
    });
  });
});
