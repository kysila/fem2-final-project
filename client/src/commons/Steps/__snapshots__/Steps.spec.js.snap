// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps component Steps render 1`] = `
<Styled(MuiBox)
  className="makeStyles-container-1"
>
  <StepsItem
    isActive={true}
    key="id-0"
    text="Cart"
  />
  <Separator
    classes={
      Object {
        "container": "makeStyles-container-1",
        "separator": "makeStyles-separator-2",
      }
    }
    key="id-0s"
  />
  <StepsItem
    isActive={false}
    key="id-1"
    src="/checkout/info"
    text="Information"
  />
  <Separator
    classes={
      Object {
        "container": "makeStyles-container-1",
        "separator": "makeStyles-separator-2",
      }
    }
    key="id-1s"
  />
  <StepsItem
    isActive={false}
    key="id-2"
    src={null}
    text="Shipping"
  />
  <Separator
    classes={
      Object {
        "container": "makeStyles-container-1",
        "separator": "makeStyles-separator-2",
      }
    }
    key="id-2s"
  />
  <StepsItem
    isActive={false}
    key="id-3"
    src={null}
    text="Payment"
  />
</Styled(MuiBox)>
`;

exports[`Steps component Steps render 2`] = `
<Styled(MuiBox)
  className="makeStyles-container-1"
>
  <StepsItem
    isActive={true}
    key="id-0"
    text="Cart"
  />
  <Separator
    classes={
      Object {
        "container": "makeStyles-container-1",
        "separator": "makeStyles-separator-2",
      }
    }
    key="id-0s"
  />
  <StepsItem
    isActive={true}
    key="id-1"
    src="/checkout/info"
    text="Information"
  />
  <Separator
    classes={
      Object {
        "container": "makeStyles-container-1",
        "separator": "makeStyles-separator-2",
      }
    }
    key="id-1s"
  />
  <StepsItem
    isActive={false}
    key="id-2"
    src={null}
    text="Shipping"
  />
  <Separator
    classes={
      Object {
        "container": "makeStyles-container-1",
        "separator": "makeStyles-separator-2",
      }
    }
    key="id-2s"
  />
  <StepsItem
    isActive={false}
    key="id-3"
    src={null}
    text="Payment"
  />
</Styled(MuiBox)>
`;

exports[`Steps component Steps render 3`] = `
<Styled(MuiBox)
  className="makeStyles-container-1"
>
  <StepsItem
    isActive={true}
    key="id-0"
    text="Cart"
  />
  <Separator
    classes={
      Object {
        "container": "makeStyles-container-1",
        "separator": "makeStyles-separator-2",
      }
    }
    key="id-0s"
  />
  <StepsItem
    isActive={true}
    key="id-1"
    src="/checkout/info"
    text="Information"
  />
  <Separator
    classes={
      Object {
        "container": "makeStyles-container-1",
        "separator": "makeStyles-separator-2",
      }
    }
    key="id-1s"
  />
  <StepsItem
    isActive={true}
    key="id-2"
    src="/checkout/shipping"
    text="Shipping"
  />
  <Separator
    classes={
      Object {
        "container": "makeStyles-container-1",
        "separator": "makeStyles-separator-2",
      }
    }
    key="id-2s"
  />
  <StepsItem
    isActive={false}
    key="id-3"
    src={null}
    text="Payment"
  />
</Styled(MuiBox)>
`;

exports[`Steps component Steps render 4`] = `
<Styled(MuiBox)
  className="makeStyles-container-1"
>
  <StepsItem
    isActive={true}
    key="id-0"
    text="Cart"
  />
  <Separator
    classes={
      Object {
        "container": "makeStyles-container-1",
        "separator": "makeStyles-separator-2",
      }
    }
    key="id-0s"
  />
  <StepsItem
    isActive={true}
    key="id-1"
    src="/checkout/info"
    text="Information"
  />
  <Separator
    classes={
      Object {
        "container": "makeStyles-container-1",
        "separator": "makeStyles-separator-2",
      }
    }
    key="id-1s"
  />
  <StepsItem
    isActive={true}
    key="id-2"
    src="/checkout/shipping"
    text="Shipping"
  />
  <Separator
    classes={
      Object {
        "container": "makeStyles-container-1",
        "separator": "makeStyles-separator-2",
      }
    }
    key="id-2s"
  />
  <StepsItem
    isActive={true}
    key="id-3"
    src="/checkout/payment"
    text="Payment"
  />
</Styled(MuiBox)>
`;
